DEVELOPING.md

```diff
@@ -1,97 +1,130 @@
 
-## Run on Android
+## Adding dependencies
 
-To run on Android, follow the instructions below for debug/release mode.
+Since we are using Yarn workspaces `dependencies` should not be added to the root `package.json` (see [more details](https://stackoverflow.com/a/53558779)).
+If multiple packages include the same dependency Yarn constraints will enforce that the version used matches.
+`devDependencies` are fine to add at the root `package.json` (see [more details](https://github.com/lerna/lerna/issues/1079#issuecomment-337660289))
+if they are shared among multiple packages, otherwise add them to just the individual package that requires it.
 
-### Debug mode
+## Testing changes during development
 
-`yarn android`
+From the root of the project you can lint and test all packages with:
 
-### Release mode
+```bash
+yarn lint
+yarn test
+```
 
-`yarn android --variant=release`
+## Manual integration testing
 
-## Run on iOS
+The [example app](examples/react-native-test-suite/) allows you to test out the React Native Embrace SDK in a representative app.
 
-To run on iOS, follow the instructions below for debug/release mode.
+Code changes will **NOT** automatically be picked up by the example app. You **MUST** follow the instructions to test out local changes.
 
-`pushd ios; pod install; popd;`
+To get started generate a new artifact from whichever packages you modified:
 
-### Debug mode
+```bash
+cd packages/<package-modified>/
+yarn build
+yarn pack
+```
 
-`yarn ios`
+Then update the example app with that local artifact:
 
-### Release mode
+```bash
+cd examples/react-native-test-suite
+yarn add ../<path-to-local-artifact>/
+yarn
+```
 
-`yarn ios --configuration=Release`
+If you alter the node_modules in any way (either by deleting them or running `yarn install`) or make further changes
+then you will need to follow the above instructions again.
 
-# Testing SDK changes locally
+Then run the app on either Android or iOS, note that React Native behaves substantially different between debug/release
+mode, so we should always test release candidates against a release build.
 
-These instructions are temporary & subject to change while we figure out a better way of testing changes.
+Android:
 
-If you alter the node_modules in any way (either by deleting them or running `yarn install`) then you might need to follow the instructions again.
+```bash
+`yarn android --variant=release`
+```
 
-## Testing JavaScript changes
+iOS:
 
-You can test React Native changes adding this library using `yarn` or `npm`. To do this you have to run `yarn pack` to generate the dist folder and then, in your test project run `yarn add localPath/embrace-io`. (If for some reason your local sdk has node-modules inside, that might bring some errors, delete that folder since it shouldn't have one)
+```bash
+pushd ios; pod install; popd;
+yarn ios --configuration=Release
+```
 
-## Testing Android changes
+## Testing against new Embrace Android SDK versions
```

Comment:
@TheMutsi same question for these Testing against new Embrace Android/iOS SDK versions sections, are the steps still what you currently do?

Reply:
Yes, they are.